src/main/java/org/jahia/modules/jahiaoauth/impl/JahiaOAuthServiceImpl.java

```diff
@@ -66,6 +80,71 @@ public JahiaOAuthServiceImpl(Map<String, DefaultApi20> oAuthDefaultApi20Map) {
         }
     }
 
+    public void setSettingsService(SettingsService settingsService) {
+        this.settingsService = settingsService;
+    }
+
+    public void setConfigurationAdmin(ConfigurationAdmin configurationAdmin) {
+        this.configurationAdmin = configurationAdmin;
+    }
+
+    public void init() {
+        BundleContext bundleContext = FrameworkService.getBundleContext();
+        //Allows to configure connector without having to go on the settings UI
+        serviceTracker = new ServiceTracker<>(bundleContext, "org.jahia.modules.jahiaoauth.service.OAuthConnectorService",
+                new ServiceTrackerCustomizer<Object, Object>() {
+                    @Override
+                    public OAuthConnectorService addingService(ServiceReference serviceReference) {
+                        OAuthConnectorService oAuthConnectorService = (OAuthConnectorService) bundleContext.getService(serviceReference);
+                        loadConnector(oAuthConnectorService,
+                                serviceReference.getProperty(JahiaAuthConstants.CONNECTOR_SERVICE_NAME).toString());
+
+                        return oAuthConnectorService;
+                    }
+
+                    @Override
+                    public void modifiedService(ServiceReference serviceReference, Object o) {
+                        OAuthConnectorService oAuthConnectorService = (OAuthConnectorService) bundleContext.getService(serviceReference);
+                        loadConnector(oAuthConnectorService,
+                                serviceReference.getProperty(JahiaAuthConstants.CONNECTOR_SERVICE_NAME).toString());
+                    }
+
+                    @Override
+                    public void removedService(ServiceReference serviceReference, Object o) {
+                    }
+                });
+        serviceTracker.open();
+    }
+
+    public void destroy() {
+        serviceTracker.close();
+    }
+
+    public void loadConnector(OAuthConnectorService oAuthConnectorService, String connectorName) {
+        Configuration[] configurations = null;
```

Comment:
Addind some logs here mentionning that a service have been detected "added or modified" and we will check for the available settings using the connector would help debug and have some info on what happens and what will be done.